### DISCUSSION TITLE
v1.1.7 Beta 4

### DISCUSSION THREAD

**Discussion by RinLit-233-shiroko** [Class-Widgets/Class-Widgets#84] [公告]
## Class Widgets 新版本！（v1.1.7-Beta）
![Banner](https://github.com/user-attachments/assets/24cec30c-a67d-4f1c-9657-76352bf83a76)

> [!IMPORTANT]
> 当前为 Beta 版本，可能存在一些不确定的 Bug，请依据情况更新。
> 
> **若您还在用旧版本 Class Widgets（≤v1.16.5），请将里面的配置文件` .\config\schedule`导出并删除目录下的文件，更新后再重新导入！**

> [!NOTE]
> 目前Linux版本 #175 可在[最新构建](https://github.com/Class-Widgets/Class-Widgets/actions/runs/12533900396)中下载，
> 但兼容性仍存在问题 #192。若需要使用建议自行使用源代码打包。

### 新增功能
- 改用PyQt5，现已**完全支持Windows7**！ #44 #80 
- 支持加载_Python语言编写_的**插件**和插件广场！#145 #164 
- 增加**异常捕获**功能，程序崩溃后将不再退出并弹出错误报告；
- 上下课铃声可分别控制启用状态；
- 记不清了（；´д｀）ゞ

#### 优化
- 修改“调休与换课”打开方式为**右键（触屏设备上为长按）**”当前活动“小组件打开；
- 在错误配置时间线并保存时增加报错提示； #184 
- 支持热切换主题
- 在滚动场景下支持触屏滑动 #85 #190 
- 为上下课提醒的背景颜色应用为自定义的颜色
- 更改自动隐藏逻辑
#### 修复
- 修复浮窗在部分设备上显示异常的问题； #201 #111 #120 
- 修复在特定情况下时间节点切换前后小组件会出现显示错误的问题； #109 
- 修复了清除日志时弹出报错的问题 #131 
- 修复了重复点击”调休与换课“中”保存“按钮后原课程表被损坏的问题 #150 
- 禁用上下课提醒后，活动开始前后或放学时无法自动切换程序主题色 #48 
- 修复其他已知问题。

<details>
<summary><strong>查看过去的更新日志（Beta版）</strong></summary>

### 新增功能
- 多天气API支持（现已支持腾讯天气、和风天气及高德天气） #90 
- 新增“浮窗”功能↘ #57 
- 为单击隐藏和自动隐藏添加了默认的隐藏方式（即可自定义隐藏行为，可在“设置”->“高级选项”中调整）
- 模糊倒计时（可在“设置”->“自定义”中设置）
- 支持带图片的自定义主题（目前试添加了2个主题：砂狼白子、星野）
- 上下课提醒、预备铃可单独启用和设置置顶 #49 

### Bug 修复
- 修复了在节点中最后一项活动为“课间”时，到达该时间点崩溃的问题 #107 
- 修复了“最大化自动隐藏”功能无法检测全屏窗口的问题 #92 
- 优化了小组件的隐藏/显示打断动画
- 修复了小组件状态更新不即时的问题 #103 
- 优化了在各个DPI的屏幕上“设置”窗口的尺寸问题 #94 

### 新增功能
- 添加了清除日志的功能
- 添加了禁用日志的选项 (可在“设置”->“高级选项”中找到)
- 在“配置文件”页面中添加了在资源管理器打开课表文件夹的按钮

### Bug 修复
- 修复了因解码问题导致某些设备启动崩溃的问题 #86
- 修复了天气小组件在无网络或网络异常情况下显示内容异常的问题

### 新增功能

- 改用 Python3.8.10 打包，现已支持 Windows 7。 #44 

#### 课程表
- 新增“节点功能”，能自由添加时段！ #32 
- 优化“时间线编辑”的设置界面
- 新增[“快速添加课程”](https://www.yuque.com/rinlit/class-widgets_help/oesr1ay7qv6fgm7w#LcseA)于“课程表编辑”，优化课表录入效率 #63 

#### 其他功能
- 新增“提醒”页面，可从此页设置“提醒”
- 新增 **“预备铃”** 功能，可在“设置”->“提醒”中设置
- 新增提醒音量滑块，可在“设置”->“提醒”中设置
- 新增“启用强调特效”选项，可以选择启用或关闭上下课提醒的水波效果
- 新增[“天气”小组件](https://www.yuque.com/rinlit/class-widgets_help/qyly70ht1ogge1pi#CbUrH)，可在“设置”->“自定义”中设置
- 新增“帮助”页面，可直接从“设置”中访问 [Class Widgets 官方文档](https://www.yuque.com/rinlit/class-widgets_help?#)
- 新增“完全显示/隐藏”于托盘菜单中，可以暂时完整隐藏小组件
- 现可通过单击 Class Widgets 托盘图标进行隐藏/显示操作

#### 界面
- 新增 **“主题”** 系统和 浅/深色选择 ，可在“设置”->“自定义”中选择 #12 

### Bug 修复

- 修复了“自定义倒计时”天数不完全准确的问题 #82 
- 修复了“生物”课程图标缺失的问题 #60 
- 修复了更改课程表文件名称并保存课程表更改后程序崩溃的问题 #59 
- 修复了上下课提醒弹窗中详细信息的错误显示 #61 
- 修复了设置时差偏移后，每日上/下午首个活动会重复提示活动开始，同时活动倒计时出错的问题 #47 
- 修复了“允许程序多开”说明文字空缺的问题 #55 
- 修复了防程序多开无作用的问题
- 修复了其他已知问题（记不清了）……
</details>

若有 Bug 或建议，欢迎在 GitHub 或 Bilibili 反馈！

**Comment by XPF705** [comment]
下载v1.1.7测试版后解压运行ClassWidgets.exe后软件并无反应，已重启电脑。电脑版本windows11专业版22H2

**Reply by RinLit-233-shiroko** [reply]
最新的v1.1.7-b2已更新，修复了由于编码原因导致的无反应崩溃。 #86 
请问在最新版本还有这个问题吗

**Reply by XPF705** [reply]
> 最新的v1.1.7-b2已更新，修复了由于编码原因导致的无反应崩溃。 #86 请问在最新版本还有这个问题吗

b2版本下载后似乎没有.exe运行文件
![image](https://github.com/user-attachments/assets/0a7d61a3-8564-4f38-9394-6d0b63e1ad48)

**Reply by RinLit-233-shiroko** [reply]
![image](https://github.com/user-attachments/assets/57c36ecc-82e8-4b6e-b5ec-5b1bd9d1fbed)
有的啊

**Reply by XPF705** [reply]
> ![image](https://private-user-images.githubusercontent.com/177950371/374532508-57c36ecc-82e8-4b6e-b5ec-5b1bd9d1fbed.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjgzODY4ODcsIm5iZiI6MTcyODM4NjU4NywicGF0aCI6Ii8xNzc5NTAzNzEvMzc0NTMyNTA4LTU3YzM2ZWNjLTgyZTgtNGI2ZS1iNWVjLTViMWJkOWQxZmJlZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAwOFQxMTIzMDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04NDBiZWZkYWE5YzNiMWEwZGQ1MzJkY2QwZmIzNmI2ZDA3Yzg2Y2I3Njk1OTk2YmVhNmQ3NTlhMjk1YzA5YWRmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.hXutcBS3t0FIQtacGL8OLcYsFXoz6Lbu0EBLDd9iK-s)有的啊

可以啦，我去学校使用几天给你反馈，感谢

**Reply by RinLit-233-shiroko** [reply]
感谢您的支持！

**Comment by XPF705** [comment]
不好意思打扰您，这是我在使用中发现的一些小问题，如有不对的请勿怪罪
1、在时间线编辑时，如果我需要设置下午的时间线，他会自动跳回上午的时间，就导致我在设置晚托和下午时相比较上午需要多一个步骤，将时段切换为晚托或者下午
2、在此我还发现一个有点奇怪的bug，在点击课程表编辑时，如果我使用快速添加课程他会闪退。
3、大多数学校在周一会进行升旗仪式，但如果这个单独进行编辑就显得有些繁琐，希望能添加对应的选项
4、关于天气板块，个人觉得ui界面有些怪怪的，希望能进行优化
5、对于置顶板块，在窗口最大化时进行收起确实能够避免放映ppt时防止遮挡又防止上课一直收着看不到时间。但在收起时还是会露一点在外面，有时还是会遮挡ppt
6、对于很多新手，希望出个教程，还是有很多人第一次使用并不知道时间线如何编辑等
不过，还是感谢您做出这么优秀的软件。第二点问题视频如下，谢谢。

Uploading 屏幕录制 2024-10-08 220042.mp4…

**Reply by RinLit-233-shiroko** [reply]
> 不好意思打扰您，这是我在使用中发现的一些小问题，如有不对的请勿怪罪 1、在时间线编辑时，如果我需要设置下午的时间线，他会自动跳回上午的时间，就导致我在设置晚托和下午时相比较上午需要多一个步骤，将时段切换为晚托或者下午 2、在此我还发现一个有点奇怪的bug，在点击课程表编辑时，如果我使用快速添加课程他会闪退。 3、大多数学校在周一会进行升旗仪式，但如果这个单独进行编辑就显得有些繁琐，希望能添加对应的选项 4、关于天气板块，个人觉得ui界面有些怪怪的，希望能进行优化 5、对于置顶板块，在窗口最大化时进行收起确实能够避免放映ppt时防止遮挡又防止上课一直收着看不到时间。但在收起时还是会露一点在外面，有时还是会遮挡ppt 6、对于很多新手，希望出个教程，还是有很多人第一次使用并不知道时间线如何编辑等 不过，还是感谢您做出这么优秀的软件。第二点问题视频如下，谢谢。
> 
> **Uploading 屏幕录制 2024-10-08 220042.mp4…**

1. 感谢反馈。这个Bug已经被修复了；
2. 第二条的视频您似乎并未上传完成，我想这个Bug应该是在未添加任何时间线的情况下点击“快速添加课程”时会崩溃吧，目前已经修复了；
3. 好的，后面我可能会让用户自定义常用的科目；
4. 这个的话欢迎来分享你的看法，这是我更改了很多次的设计了（）；
5. 好的收到，会添加自定义选项的；
6. 这个现在已经实现了，可在“设置”->“帮助”页找到我们的使用文档。如果您需要视频教程的话，后面会考虑制作的

**Comment by XPF705** [comment]
您好，感谢您之前对我留言的回应，在重新使用最新版本时，以上问题均已被解决。但同时在使用过程中我也想再提出一些小建议
1、预备铃提醒，是个非常实用的功能，但是否能与上下课提醒分开呢？有时并不需要用到上下课提醒而是听广播即可。况且一下课准时弹出这个会让教师感到反感，所以希望与上下课提醒分开。

**Reply by RinLit-233-shiroko** [reply]
好。但是对于第一点你的意思是在某些课程不启用预备铃吗，还是给预备铃增加开关选项，如果是第二个，那你可以在”提醒“中将预备铃提前分钟数调为”0“即可。

**Reply by XPF705** [reply]
> 好。但是对于第一点你的意思是在某些课程不启用预备铃吗，还是给预备铃增加开关选项，如果是第二个，那你可以在”提醒“中将预备铃提前分钟数调为”0“即可。

不是的，我的意思是现在是只能选择开启全部提示而不能单独只开预备铃。比如我只需要提醒预备铃但不需要上下课提醒。

**Reply by RinLit-233-shiroko** [reply]
好的，收到

**Reply by RinLit-233-shiroko** [reply]
> > 好。但是对于第一点你的意思是在某些课程不启用预备铃吗，还是给预备铃增加开关选项，如果是第二个，那你可以在”提醒“中将预备铃提前分钟数调为”0“即可。
> 
> 不是的，我的意思是现在是只能选择开启全部提示而不能单独只开预备铃。比如我只需要提醒预备铃但不需要上下课提醒。

目前已经在Beta3实现了！

**Reply by XPF705** [reply]
> > > 好。但是对于第一点你的意思是在某些课程不启用预备铃吗，还是给预备铃增加开关选项，如果是第二个，那你可以在”提醒“中将预备铃提前分钟数调为”0“即可。
> > 
> > 
> > 不是的，我的意思是现在是只能选择开启全部提示而不能单独只开预备铃。比如我只需要提醒预备铃但不需要上下课提醒。
> 
> 目前已经在Beta3实现了！

好的！！！

**Comment by baiyao105** [comment]
在使用发现了一些缺点,希望改进
建议:
1.学校的预备铃存在与课程一样的多个时间偏差,希望调整精度和加入多个预备铃时间/开关某些课程的预备铃
2.希望在添加课程时可以自定义/修改原来课表的课程显示图标
3.大课间环境是复杂的,是否可以定义一些课间单独为大课间,而不是设置课程
4.希望添加预备铃提醒铃声单独开关的功能
5.自定义倒计时根据时间距离,修改为小时/分钟做单位,并希望提升自定义倒计时的精确时间
6.希望将靠近的(连在一起)时间节点显示到同一个"接下来",
7.建议加入同步课表功能开关,填写自己的json下载地址(看起来会方便许多?)(后续功能也可以用?)
8.作为一个课程表,也许可以加入公告的功能,可以统计成绩?(bushi)(
9.添加报时的置顶弹窗,课程剩余用户自定义时间时置顶弹窗提示(上下课样式),并添加特效和声音开关(如果开启会导致教师反感)
问题:
1.我的电脑软件缩放比例为100%,如果进行提前配置带到学校150%缩放会出现组件的缩放问题(超出屏幕),希望添加自适应软件的缩放比例
[并且如果在改为正常可显示缩放比例或小尺寸模式,使用自定义功能删除或移动组件位置,都会导致整个组件向上超出屏幕外]
2.在使用在窗口最大化时收起,会导致设置失败(不生效,不保存)
3.两个时间点过近时会出现上个时间点的课程倒计时变成下个时间的30分钟开始倒计时,希望优化
(此时上一节点的最后一个课程正在进行,出现当前活动“暂无课程”，活动倒计时“距离上课还有 --:--”,下课与课程均会受到影响〕

**Comment by InnFen** [comment]
希望能加入毛玻璃效果！

**Comment by XPF705** [comment]
你好，在班级深度使用半个月后，我提出以下意见
1、希望增加设置体育课或其他课程自动关机。
2、希望增加倒计时时间滚动效果，就是数字从上出现，从下消失的滚动效果
3、能否做成软件？

**Reply by MKStoler4096** [reply]
我个人看法如下（不代表他人立场）：
1. 自动化的话可以先去看看`ClassIsland`。
2. 不好说，本来在低配机上就很卡了。
3. 本来不就是软件吗🤣

**Reply by Detritalw** [reply]
@MKStoler4096 第三点他应该说的是安装程序

**Reply by pizeroLOL** [reply]
第一个需求完全可以考虑学习一下windows的计划任务

**Comment by suifeng333** [comment]
最多只能添加9个时间点，添加到第9个不能再添加
